src/Features/VisualBasic/Portable/ChangeSignature/VisualBasicChangeSignatureService.vb

```diff
@@ -353,7 +360,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ChangeSignature
 
             If vbnode.IsKind(SyntaxKind.Attribute) Then
                 Dim attribute = DirectCast(vbnode, AttributeSyntax)
-                Dim newArguments = PermuteArgumentList(attribute.ArgumentList.Arguments, updatedSignature, declarationSymbol)
+
+                Dim semanticModel = Await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(False)
+                Dim symbolInfo = semanticModel.GetSymbolInfo(DirectCast(originalNode, AttributeSyntax))
+                Dim methodSymbol = TryCast(symbolInfo.Symbol, IMethodSymbol)
+
+                Dim newArguments = PermuteArgumentList(attribute.ArgumentList.Arguments, updatedSignature.WithoutAddedParameters(), declarationSymbol)
+                newArguments = AddNewArgumentsToList(methodSymbol, newArguments, updatedSignature, isReducedExtensionMethod:=False, isParamsArrayExpanded:=False, generateAttributeArguments:=True)
```

Comment:
i feel like i say this same logic repeated multiple times here and in C#.

Reply:
The Permute/Add logic has been deduped within each language, but not across languages.